### DISCUSSION TITLE
Websocket analyzer

### DISCUSSION THREAD

**Discussion by jgvt** [zeek/zeek#1637] [Ideas]
Hi everyone,

After looking inside HTTP analyzer code, I saw that after reply 101 (websocket upgrade) there is no analyze on the flow.

Is it a good idea to add a websocket analyzer ?
We can even add PIA analyzer on websocket to activate dpd (to prevent evading by tunneling into websocket flow) and create special signature on websocket (ex: write signatures for malware like Drovorub).

What do you think about that ?

**Comment by rsmmr** [comment]
Sounds great to me. There's some prior work using Spicy: https://collateralmeaning.blogspot.com/2017/10/bro-and-spicy-for-websockets-network.html (note that was done with an old version of Spicy).

**Reply by jgvt** [reply]
Find all files of this article here: https://github.com/jennifergates/paper
I already wrote some stuff in binpac but I will try to use Spicy instead.

I'm asking myself about the best way to achieve this.
Does he have to be son of HTTP analyzer (to avoid HTTP header treatment) ?

**Comment by jgvt** [comment]
Should I create a new package rather than directly integrate inside Zeek like http analyzer ?

**Reply by 0xxon** [reply]
A package is probably the way to go - especially if you use spicy (which is separate from Zeek - so it can actually not be shipped with Zeek).

**Reply by jgvt** [reply]
I have already a Binpac version of this analyzer. Should I push this one first on zeek and work on a second write in Spicy and push on packages ?
